### PR TITLE
Fixes #1675 - T1 eexec end finding spec compliance

### LINKED ISSUE #1675: fontTools.t1Lib.T1Error: can't find end of eexec part
The T1 font parser is unable to handle some fonts (which conform to the spec). It complains:

```python
>>> f = T1Font('some-t1-font')
fontTools.t1Lib.T1Error: can't find end of eexec part
```
In particular, this happens with (spec-conformant) fonts that don't have 64 consecutive ASCII `0`s immediately after the `eexec` section:

```bash
$ tail -c 550 some-t1-font | xxd
00000000: 0a86 c365 74ba 6d9f 36e9 3030 3030 3030  ...et.m.6.000000
00000010: 3030 3030 3030 3030 3030 3030 3030 3030  0000000000000000
00000020: 3030 3030 3030 3030 3030 0a30 3030 3030  0000000000.00000
00000030: 3030 3030 3030 3030 3030 3030 3030 3030  0000000000000000
00000040: 3030 3030 3030 3030 3030 300a 3030 3030  00000000000.0000
00000050: 3030 3030 3030 3030 3030 3030 3030 3030  0000000000000000
00000060: 3030 3030 3030 3030 3030 3030 0a30 3030  000000000000.000
00000070: 3030 3030 3030 3030 3030 3030 3030 3030  0000000000000000
00000080: 3030 3030 3030 3030 3030 3030 300a 3030  0000000000000.00
00000090: 3030 3030 3030 3030 3030 3030 3030 3030  0000000000000000
000000a0: 3030 3030 3030 3030 3030 3030 3030 0a30  00000000000000.0
000000b0: 3030 3030 3030 3030 3030 3030 3030 3030  0000000000000000
000000c0: 3030 3030 3030 3030 3030 3030 3030 300a  000000000000000.
000000d0: 3030 3030 3030 3030 3030 3030 3030 3030  0000000000000000
000000e0: 3030 3030 3030 3030 3030 3030 3030 3030  0000000000000000
000000f0: 0a30 3030 3030 3030 3030 3030 3030 3030  .000000000000000
00000100: 3030 3030 3030 3030 3030 3030 3030 3030  0000000000000000
00000110: 300a 3030 3030 3030 3030 3030 3030 3030  0.00000000000000
00000120: 3030 3030 3030 3030 3030 3030 3030 3030  0000000000000000
00000130: 3030 0a30 3030 3030 3030 3030 3030 3030  00.0000000000000
00000140: 3030 3030 3030 3030 3030 3030 3030 3030  0000000000000000
00000150: 3030 300a 3030 3030 3030 3030 3030 3030  000.000000000000
00000160: 3030 3030 3030 3030 3030 3030 3030 3030  0000000000000000
00000170: 3030 3030 0a30 3030 3030 3030 3030 3030  0000.00000000000
00000180: 3030 3030 3030 3030 3030 3030 3030 3030  0000000000000000
00000190: 3030 3030 300a 3030 3030 3030 3030 3030  00000.0000000000
000001a0: 3030 3030 3030 3030 3030 3030 3030 3030  0000000000000000
000001b0: 3030 3030 3030 0a30 3030 3030 3030 3030  000000.000000000
000001c0: 3030 3030 3030 3030 3030 3030 3030 3030  0000000000000000
000001d0: 3030 3030 3030 300a 3030 3030 3030 3030  0000000.00000000
000001e0: 3030 3030 3030 3030 3030 3030 3030 3030  0000000000000000
000001f0: 3030 3030 3030 3030 0a30 3030 3030 3030  00000000.0000000
00000200: 3030 3030 3030 3030 3030 3030 3030 3030  0000000000000000
00000210: 3030 3030 3030 3030 300a 636c 6561 7274  000000000.cleart
00000220: 6f6d 6172 6b0a                           omark.
```

The problem is that this file has runs of 32 ASCII `0`'s separated by line feeds. According to [the spec](https://www-cdf.fnal.gov/offline/PostScript/T1_SPEC.PDF) (page 14) this is valid:

> The text encrypted by **eexec** must be followed by 512 ASCII zeros. There may be white space characters (blank, tab, carriage return or line feed) interspersed among these zeros.

The culprit here is:

https://github.com/fonttools/fonttools/blob/f9d12be58fde76c5f8bbd8ac207223874494c713/Lib/fontTools/t1Lib/__init__.py#L276

https://github.com/fonttools/fonttools/blob/f9d12be58fde76c5f8bbd8ac207223874494c713/Lib/fontTools/t1Lib/__init__.py#L308-L329

A contiguous run of 64 ASCII `0`s is expected. But it would be to spec to have `0\n0\t0\r0 0` etc. trailing the eexec section.

I have several different T1 fonts that exhibit this behavior (one such font is New Century Schoolbook--although I don't want to get you DMCA'd by uploading it). They can all be read with macOS Font Book and FontForge. And indeed, if I change `EEXECEND = b'0' * 64` to `EEXECEND = b'0' * 32`, fontTools successfully parses the font and is capable of drawing the glyphs.

### PR DESCRIPTION
**Pull request by baileyparker**
The T1 spec (https://www-cdf.fnal.gov/offline/PostScript/T1_SPEC.PDF) on
page 14 says that:

> The text encrypted by eexec must be followed by 512 ASCII zeros. There may be
> white space characters (blank, tab, carriage return or line feed)
> interspersed among these zeros.

  * Add a failing test case with white space characters interspersed
    among the trailing zeros after the eexec

  * Fix eexec end finding so that it passes this new test case

This modification to the method for finding the end of the eexec section is more robust than the current method (which looks for 64 consecutive ASCII `0`s immediately after the encrypted text ends) as it handles the spec's allowance for arbitrary interspersed whitespace. A regex was convenient here (and was already used in `findEncryptedChunks`), but it could be unrolled into a function if need be.

While more robust than the existing function, it's unclear to me how the spec would handle an eexec section that coincidentally has 512 ASCII `0`s in the middle of it (or any combination of that with interspersed whitespace). This approach does not handle that, but neither does the existing solution.

### CODE REVIEW

File: Lib/fontTools/t1Lib/__init__.py [hunk 1]
```diff
@@ -312,9 +314,10 @@ def findEncryptedChunks(data):
 		if eBegin < 0:
 			break
 		eBegin = eBegin + len(EEXECBEGIN) + 1
-		eEnd = data.find(EEXECEND, eBegin)
-		if eEnd < 0:
+		endMatch = EEXECEND.search(data, eBegin)
+		if not endMatch:
 			raise T1Error("can't find end of eexec part")
+		eEnd = endMatch.start()
```
Comment (justvanrossum) [inline Lib/fontTools/t1Lib/__init__.py]: On second though, I have one minor change request:
The idiomatic way to check for None is `if endMatch is None: ...`

Comment (baileyparker) [reply Lib/fontTools/t1Lib/__init__.py]: Sure! Done. :smile:

**Comment by baileyparker** [comment]
Thanks for addressing this so quickly 😁

**Comment by justvanrossum** [comment]
Thanks for your contribution! 👍

**Pull request merged**